test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+        Integer key = getKeyID();
+        String value1 = getValue();
+
+        Version version1 = newKeyValueTable.put(null, key, value1).join();
+        assertNotNull("Vesrion1 should not be null", version1);
+        log.info("version1: " + version1);
+        TableEntry<Integer, String> tableEntry1 = newKeyValueTable.get(null, key).join();
+        log.info("Get tableEntry1: {}", tableEntry1);
+        assertEquals("Corresponding key for entry1 should be as inserted",
+                key, tableEntry1.getKey().getKey());
+        assertEquals("Corresponding value for entry1 should be as inserted",
+                value1, tableEntry1.getValue());
+
+        String value2 = getValue();
+        Version version2 = newKeyValueTable.put(null, key, value2).join();
+        assertNotNull("Vesrion2 should not be null", version1);
+        log.info("version2: " + version2);
+        TableEntry<Integer, String> tableEntry2 = newKeyValueTable.get(null, key).join();
+        log.info("Get tableEntry2: {}", tableEntry2);
+        assertEquals("Corresponding key for entry2 should be as inserted",
+                key, tableEntry2.getKey().getKey());
+        assertEquals("Corresponding value for entry2 should be as inserted",
+                value2, tableEntry2.getValue());
+        assertNotEquals("Corresponding value for entry2 should not remian as entry1",
+                value1, tableEntry2.getValue());
+        assertNotEquals("Version1 and Version2 should be different in values", version1, version2);
+
+        AssertExtensions.assertSuppliedFutureThrows(
+                "Table entry with an older key version should not get removed",
+                () -> newKeyValueTable.remove(null, key, version1),
+                ex -> ex instanceof BadKeyVersionException
+        );
+        log.info("Found BadKeyVersionException while removing the table entry with an older key version");
+
+        assertNull("No error should be obtained during key remove",
+                newKeyValueTable.remove(null, key, version2).join());
+        log.info("The requested table entry with specified key & its version is removed");
+
+        assertNull("Null should be returned as the reqested key entry is already removed",
+                newKeyValueTable.get(null, key).join());
+        log.info("Null obtained on requsting to fetch a key entry which does not exists");
+
+        AssertExtensions.assertSuppliedFutureThrows(
+                "Error should be obtained on attempt to delete already removed key entry",
+                () -> newKeyValueTable.remove(null, key, version2),
+                ex -> ex instanceof NoSuchKeyException
+        );
+        log.info("Found NoSuchKeyException on attempt to delete already removed key entry");
+        log.info("Successfully completed test Conditional delete single Key using its version");
+    }
+
+    /**
+     * Test case-46: Get Keys version List
+     */
+    @Test
+    public void testGetKeyVersionList() {
```

Comment:
This is already covered in the base class.

Reply:
Sure, I'll remove this test as its a duplicate.